spectrocloud/resource_ssh_key.go

```diff
@@ -0,0 +1,125 @@
+package spectrocloud
+
+import (
+	"context"
+	"encoding/base64"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/spectrocloud/hapi/models"
+	"github.com/spectrocloud/palette-sdk-go/client"
+	"time"
+)
+
+func resourceSSHKey() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceSSHKeyCreate,
+		ReadContext:   resourceSSHKeyRead,
+		UpdateContext: resourceSSHKeyUpdate,
+		DeleteContext: resourceSSHKeyDelete,
+		Description:   "A resource for creating and managing ssh keys.",
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		SchemaVersion: 2,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the ssh key. This name is used to identify the ssh key in the Spectro Cloud UI.",
+			},
+			"ssh_key": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+				StateFunc: func(val any) string {
+					return base64.StdEncoding.EncodeToString([]byte(val.(string)))
+				},
+				Description: "The SSH key to be used for the cluster. This is the public key that will be used to access the cluster.",
+			},
+			"context": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "project",
+				ValidateFunc: validation.StringInSlice([]string{"", "project", "tenant"}, false),
+				Description:  "Context of the ssh key. This can be either project or tenant. If not specified, the default value is project.",
```

Comment:
```suggestion
				Description:  "Context of the ssh key. This can be either project or tenant. If not specified, the default value is `project`.",
```